reference/enterprise-api-ref/build-api.markdown

```diff
@@ -0,0 +1,829 @@
+---
+layout: default
+title: Build API
+published: true
+tags: [reference, enterprise, API, build, modules]
+---
+
+The Build API enables you to easily manage policy projects and their respective CFEngine Build modules.
+
+# Projects API
+
+A project is a set of CFEngine Build modules and custom files/json/policy files.
+
+## Create project
+
+**URI:** https://hub.cfengine.com/api/build/projects
+
+**Method:** POST
+
+**Parameters:**
+
+* **repositoryUrl** *(string)*
+  Git repository URL. Project will be synchronized with this repository. Supported protocols: `http`, `https`, `ssh`
+  , `git`. Required. Git repository URL example: https://github.com/cfengine/build-index.git
+
+* **branch** *(string)*
+  Repository branch. Required.
+* **name** *(string)*
+  Project name. Required.
+* **authenticationType** *(string)*
+  Authentication type that will be used to get access to the repository. Allowed values: `password`, `ssh_key`.
+  Required.
+* **username** *(string)*
+  Username for authentication to the repository. Required when authentication type is `password`.
+* **password** *(string)*
+  Password for authentication to the repository. Required when authentication type is `password`.
+* **sshPrivateKey** *(string)*
+  SSH private key for authentication to the repository. Required when authentication type is `ssh_key` and `sshKeyId` is
+  not set.
+* **sshKeyId** *(integer)*
+  Generated SSH private key ID by [SSH keys API][SSH keys API#Generate SSH key] for authentication to the repository.
+  Required when authentication type is `ssh_key` and `sshPrivateKey` is not set.
+
+**Example request (curl):**
+
+```console
+curl --user <username>:<password> \
+  -X POST \
+  https://hub.cfengine.com/api/build/projects \
+  -H 'content-type: application/json' \
+  -d '
+  {
+  "repositoryUrl": "https://github.com/username/repository.git",
+  "branch": "master",
+  "authenticationType": "password",
+  "password" : "git_token_or_password",
+  "username" : "git_username",
+  "name": "Production"
+  }'
+```
+
+**Successful response example:**
+
+```
+HTTP 200 Ok
+{
+    "id": "8"
+}
+```
+
+**Responses:**
+
+| HTTP response code | Description |
+|--------|-------------|
+| 200 OK | Project successfully created |
+| 422 Unprocessable entity | Validation error occurred |
+| 500 Internal server error | Internal server error |
+
+## Update project
+
+By changing the repository url or branch you will initialize a new project and the current one will be removed from the
+file system and any un-pushed/un-deployed(terminology in Mission Portal UI) changes will be lost.
+
+**URI:** https://hub.cfengine.com/api/build/projects/:id
+
+**Method:** PATCH
+
+**Parameters:**
+
+* **id** *(integer)*
+  Project's ID. Required.
+
+* **repositoryUrl** *(string)*
+  Git repository URL. Project will be synchronized with this repository. Supported protocols: `http`, `https`, `ssh`
+  , `git`. Required. Git repository URL example: https://github.com/cfengine/build-index.git
```

Comment:
```suggestion
  , `git`. Required. Git repository URL example: https://github.com/username/repository.git
```
Build-index is a bad example, it's not the same format as what they would see in their own project.